### ISSUE TITLE
Open Capture dialogue disappears instantly on KDE

### ISSUE THREAD

**Issue by dylif** [flameshot-org/flameshot#2342]
### Flameshot Version

Flameshot v11.0.0 (1cc5a262)

### Installation Type

User repository (AUR), Compiled from source

### Operating System type and version

Arch Linux 5.16.3-arch1-1, KDE Plasma 5.23.5

### Description

When attempting to open a capture made by mouse selection in an application, the dialogue to select a program to open the capture with disappears almost instantly.

### Steps to reproduce

1. Run `flameshot gui`
2. Capture part of the screen using the mouse
3. Use Ctrl-O shortcut to open the capture in an application
4. Observe the dialogue to choose an application display for an instant then disappear

### Screenshots or screen recordings

![2022-01-30 01-23-17](https://user-images.githubusercontent.com/18730377/151689243-489dea1a-dcad-44a3-9ea6-23ead91c859f.gif)

### System Information

1. Arch Linux 5.16.3-arch1-1
2. ![image](https://user-images.githubusercontent.com/18730377/151689268-9917b5d7-8841-4716-b919-382243a4735e.png)
3. 
    - KDE Plasma 5.23.5, Kwin 5.23.5
    - Using Xorg

**Comment by mmahmoudian** [comment]
I tried to replicate this. It seems this only happens when `flameshot gui` is called. When the tray icon is clicked, this does not happen!

**Comment by dylif** [comment]
That's strange. Thank you for your comments. I will use this as a workaround for now.

**Comment by cheeseandcereal** [comment]
+1 also having this issue. Using i3 on 5.16.5-arch1-1 with the 'official' package from the community repository. flameshot v11.0.0 (package version 11.0.0-4)

**Comment by xrat** [comment]
+1 Ubuntu 20.4 with MATE and LXDE. Workaround confirmed for `flameshot launcher` (on box where I have not tray icon).

**Comment by yar85** [comment]
+1 same issue
Manjaro KDE (frameworks 5.90.0, Plasma 5.23.5), kernel 5.16.7-1, Qt (system) 5.15.2, X11. 
Flameshot 11.0.0 (-) compiled with Qt 5.15.2, package ver 11.0.0-4, installed from community repo (manjaro official).
Full `dbus-monitor` output from relaunch, hope it helps somehow:
```
signal time=1645201368.790554 sender=org.freedesktop.DBus -> destination=:1.14331 serial=2 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameAcquired
   string ":1.14331"
signal time=1645201368.790574 sender=org.freedesktop.DBus -> destination=:1.14331 serial=4 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameLost
   string ":1.14331"
method call time=1645201375.614291 sender=:1.14392 -> destination=org.freedesktop.DBus serial=16 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=AddMatch
   string "type='signal',sender='org.freedesktop.DBus',interface='org.freedesktop.DBus',member='NameOwnerChanged',arg0='org.a11y.Bus',arg1=''"
method call time=1645201375.614427 sender=:1.14392 -> destination=org.freedesktop.DBus serial=17 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameHasOwner
   string "org.a11y.Bus"
method call time=1645201375.614561 sender=:1.14392 -> destination=org.freedesktop.DBus serial=18 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=AddMatch
   string "type='signal',sender='org.freedesktop.DBus',interface='org.freedesktop.DBus',member='NameOwnerChanged',arg0='org.a11y.Bus'"
method call time=1645201375.614601 sender=:1.14392 -> destination=org.freedesktop.DBus serial=19 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=GetNameOwner
   string "org.a11y.Bus"
method call time=1645201375.614885 sender=:1.14392 -> destination=org.a11y.Bus serial=20 path=/org/a11y/bus; interface=org.a11y.Bus; member=GetAddress
method call time=1645201375.615221 sender=:1.14392 -> destination=org.freedesktop.DBus serial=21 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=AddMatch
   string "type='signal',path='/KDEPlatformTheme',interface='org.kde.KDEPlatformTheme',member='refreshFonts'"
method call time=1645201375.615241 sender=:1.14392 -> destination=org.freedesktop.DBus serial=22 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=AddMatch
   string "type='signal',path='/KToolBar',interface='org.kde.KToolBar',member='styleChanged'"
method call time=1645201375.615582 sender=:1.14392 -> destination=org.kde.StatusNotifierWatcher serial=23 path=/StatusNotifierWatcher; interface=org.kde.StatusNotifierWatcher; member=RegisterStatusNotifierItem
   string ":1.14393"
```

**Comment by mmahmoudian** [comment]
just to update, the issue persists on Flameshot v11.0.0 (aa6b5eac) on two different computers both sharing the following:
- same Manjaro kernel
- same  KDE Plasma
- X11

The hardware and software stacks are mostly different

**Comment by borgmanJeremy** [comment]
We need to refactor the launcher tool to be similar to the pin or upload widget. The problem right now is the lifetime of the dialog box is tied to the capture widget which is deleted as the capture is closed. This got missed when we refactored the way all the tools work.

**Comment by dylif** [comment]
Just want to add that this issue has been resolved as of commit e9c349b3

**Comment by mmahmoudian** [comment]
@dylif thanks for pointing this out.

**Comment by danktankk** [comment]
this is not fixed for pop-os 22.04.  I have the exact same issue where the gui call opens the window and then immediately closes.  is there a workaround or anything I can do?

Flameshot v11.0.0 (11.0.0-2 Debian)
Compiled with Qt 5.15.2

**Comment by xrat** [comment]
@danktankk upgrade to v12.
